ui/src/app/pages/demographic-survey.tsx

```diff
@@ -58,21 +62,43 @@ export const DemographicSurvey = (props: WithSpinnerOverlayProps) => {
     setChanged(!fp.isEqual(initialSurvey, profile?.demographicSurveyV2));
   }, [profile]);
 
+  // Users, who still have not filled survey v2 after 30days, will be shown the survey page after every sign in
+  // In such scenarios the location pathname will be different from DEMOGRAPHIC_SURVEY_V2_PATH
```

Comment:
interesting!  I see that you will then allow the navigation to continue after the useer completed the survey.  Very nice.